storage/innobase/buf/buf0dblwr.cc

```diff
@@ -719,10 +719,14 @@ buf_dblwr_init_or_load_pages(
 			return(DB_ERROR);
 		}
 
+		byte zero_page[UNIV_PAGE_SIZE_MAX] = {0};
 		for (page = recovery_buf; page < recovery_buf + size;
 		     page += UNIV_PAGE_SIZE) {
 
-			recv_dblwr.add(page);
+			/* Skipping all zero pages */
+			if (memcmp(page, zero_page, UNIV_PAGE_SIZE) != 0) {
```

Comment:
That's a big memcmp. Can you check the checksum field first?

Reply:
This is more like an optimization. I observed that lot of 'all-zero' are added in doublwrite recovery page list. And this iteration is of fixed size (bufferpools * 127).

Checksum can also be zero. So we cannot use it.
There is existing function buf_page_is_all_zeroes(). Since it is using for loop and byte by byte checking, I decided to use memcp.

Reply:
I mean:
check if checksum field is zero
if not zero, page is not all zeroes
if zero, memcmp the whole page

Reply:
ah, OK. Will implement this.